internal/evaluator/rego.go

```diff
@@ -118,6 +157,39 @@ func ociBlob(bctx rego.BuiltinContext, a *ast.Term) (*ast.Term, error) {
 	return ast.StringTerm(blob.String()), nil
 }
 
+func purlIsValid(bctx rego.BuiltinContext, a *ast.Term) (*ast.Term, error) {
+	uri, ok := a.Value.(ast.String)
+	if !ok {
+		return ast.BooleanTerm(false), nil
+	}
+	_, err := packageurl.FromString(uri.String())
```

Comment:
This needs to be `string(uri)` instead of `uri.String()`. The second form adds quotes. I ran into this with the fetch oci blob function as well. 

(same for the PURL parse function)